modules/solid_mechanics/src/materials/CombinedPlasticityStressUpdate.C

```diff
@@ -0,0 +1,240 @@
+#include "CombinedPlasticityStressUpdate.h"
+#include "Function.h"
+#include "ElasticityTensorTools.h"
+#include "BackstressRadialReturnStressUpdate.h"
+#include "ElasticityTensorTools.h"
+
+registerMooseObject("SolidMechanicsApp", ADCombinedPlasticityStressUpdate);
+registerMooseObject("SolidMechanicsApp", CombinedPlasticityStressUpdate);
+
+template <bool is_ad>
+InputParameters
+CombinedPlasticityStressUpdateTempl<is_ad>::validParams()
+{
+  InputParameters params = BackstressRadialReturnStressUpdateTempl<is_ad>::validParams();
+  params.addClassDescription("This class uses the discrete material in a radial return isotropic"
+                             "plasticity model.  This class is one of the basic radial return "
+                             "constitutive models, yet it can be used in conjunction with other "
+                             "creep and plasticity materials for more complex simulations.");
+  params.addParam<Real>("Q", 0.0, "Saturation value for isotropic hardening (Q in Voce model)");
+  params.addParam<Real>("b", 0.0, "Rate constant for isotropic hardening (b in Voce model)");
+  params.addParam<FunctionName>("yield_stress_function",
+                                "Yield stress as a function of temperature");
+  params.addParam<Real>("yield_stress", "The point at which plastic strain begins accumulating");
+  params.addParam<FunctionName>("isotropic_hardening_function",
+                                "True stress as a function of plastic strain");
+  params.addParam<Real>("isotropic_hardening_constant", "Isotropic hardening slope");
+  params.addCoupledVar("temperature", 0.0, "Coupled Temperature");
+  params.set<std::string>("effective_inelastic_strain_name") = "effective_plastic_strain";
+  params.addParam<Real>("kinematic_hardening_modulus", 0.0, "Kinematic hardening modulus");
+  params.addParam<Real>("material_constant_gamma",
```

Comment:
This parameter name should maybe reflect that it's for kinematic hardening. Maybe `kinematic_hardening_gamma` or something?

Reply:
Changing the name to: kinematic_saturation_gamma